src/services/correios-alt.js

```diff
@@ -0,0 +1,59 @@
+'use strict'
+
+import fetch from 'node-fetch'
+import ServiceError from '../errors/service.js'
+
+export default function fetchCorreiosAltAPIService(
+  cepWithLeftPad,
+  configurations
+) {
+  const url = 'https://buscacepinter.correios.com.br/app/cep/carrega-cep.php'
+  const options = {
+    method: 'POST',
+    mode: 'cors',
+    headers: {
+      'content-type': 'application/x-www-form-urlencoded; charset=UTF-8'
+    },
+    body: `cep=${cepWithLeftPad}`,
+    timeout: configurations.timeout || 30000
+  }
+
+  return fetch(url, options)
+    .then(parseResponse)
+    .then(extractCepValuesFromResponse)
+    .catch(throwApplicationError)
+}
+
+function parseResponse(response) {
+  return response.json().then(result => {
+    if (result.total === 0 || result.erro || result.dados[0].cep === "") {
```

Comment:
Dúvida, se por algum motivo a API parar de retornar `result.dados` como um array, será que esse método ainda vai funcionar como esperado? 🤔 

Sinto que o erro sobe e vai sim mas to aqui na dúvida

Reply:
Bom ponto, o erro vai subir sim, mas vai subir como TypeError e o catch lá vai encapsular no ServiceError.

Se deixar de retornar `dados`, vai ser algo como `TypeError: Cannot read properties of undefined (reading '0')`
Se passar a retornar outro tipo de dado vai ser algo como: `TypeError: Cannot read properties of undefined (reading 'cep')`

Eu acredito que se fizer mais checks pra garantir que o `dados` exista e seja array, vai acabar mascarando o real problema da API ter alterado o formato de retorno.

Reply:
Vc tem um ótimo ponto mestre! ❤️